zipline/protocol.py

```diff
@@ -506,6 +545,26 @@ def has_key(self, name):
         """
         return name in self
 
+    def get_default(self, name):
```

Comment:
hm. I got get_default from when I replaced the Positions defaultdict, where just using get felt wrong. Though I suppose `defaultdict.get` does it's business that way too. I want something like `setdefault` but with the hint that we will create an object in the method name and not the class type.